app/utils/IncomeSourcesUtils.scala

```diff
@@ -117,28 +149,67 @@ object IncomeSourcesUtils {
     }
   }
 
-  def getUKPropertyDetailsFromSession(implicit user: MtdItUser[_]): Either[Throwable, CheckUKPropertyViewModel] = {
-    val result: Option[Either[Throwable, CheckUKPropertyViewModel]] = for {
-      tradingStartDate <- user.session.data.get(addUkPropertyStartDate)
-      cashOrAccrualsFlag <- user.session.data.get(addIncomeSourcesAccountingMethod)
-    } yield {
-      Right(CheckUKPropertyViewModel(
-        tradingStartDate = LocalDate.parse(tradingStartDate),
-        cashOrAccrualsFlag = cashOrAccrualsFlag
-      ))
+  def getUKPropertyDetailsFromSession(sessionService: SessionService)(implicit user: MtdItUser[_], ec: ExecutionContext): Future[Either[Throwable, CheckUKPropertyViewModel]] = {
+    for {
+      startDate <- sessionService.get(addUkPropertyStartDate)
+      accMethod <- sessionService.get(addIncomeSourcesAccountingMethod)
+    } yield (startDate, accMethod) match {
+      case (Right(dateMaybe), Right(methodMaybe)) =>
+        val maybeModel = for {
+          foreignPropertyStartDate <- dateMaybe.map(LocalDate.parse)
+          cashOrAccrualsFlag <- methodMaybe
+        } yield {
+          CheckUKPropertyViewModel(
+            tradingStartDate = foreignPropertyStartDate,
+            cashOrAccrualsFlag = cashOrAccrualsFlag)
+        }
+        maybeModel.map(Right(_))
+          .getOrElse(Left(new Error("Unable to construct model")))
```

Comment:
if you can bee specific in the comment, what exact model we failed to construct?